src/js_of_ocaml_rules.ml

```diff
@@ -19,22 +19,29 @@ let install_jsoo_hint = "try: opam install js_of_ocaml-compiler"
 let in_build_dir ~ctx args =
   Path.L.relative ctx.Context.build_dir (".js" :: args)
 
+let jsoo sctx =
+  SC.resolve_program sctx ~loc:None ~hint:install_jsoo_hint "js_of_ocaml"
+
 let runtime_file ~sctx file =
   match
     Artifacts.file_of_lib (SC.artifacts sctx)
       ~loc:Loc.none
       ~lib:(Lib_name.of_string_exn ~loc:None "js_of_ocaml-compiler") ~file
   with
   | Error _ ->
-    Arg_spec.Dyn (fun _ ->
-      Utils.library_not_found ~context:(SC.context sctx).name
-        ~hint:install_jsoo_hint
-        "js_of_ocaml-compiler")
+    begin match jsoo sctx with
+    | Ok path when Path.exists (Path.relative path file) ->
```

Comment:
This shouldn't be a static check, otherwise it won't work when js_of_ocaml is part of the workspace. Instead, you can use `Build.if_file_exists`.

Reply:
Sorry, coming back to this after some time. I tried to follow you suggestion, so that the type of the function `Js_of_ocaml.runtime_file` becomes `(unit, 'a Arg_spec.t) Build.t` and return the path of the runtime file as a program argument.

This function is consumed as follows:
```

Reply:
The idea is to add the result inside the arrow, and then you can get it inside the argument list with `Dyn`:

```
(Build.arr (fun x -> x) &&& runtime)
>>> 
Build.run ~dir
  jsoo
  [ Dyn (fun (flags, runtime) -> flags @ [runtime])
  ; ...
  ]
```

Reply:
Thanks! That put me in the right direction.